.github/workflows/build_rocks.yaml

```diff
@@ -96,69 +107,85 @@ jobs:
+                            name: rockName,
+                            version: rockVersion,
+                            path: rockPath,
+                            arch: arch,
+                            image: image,
+                            "rockcraft-revision": rockcraftRevisions[arch] || '',
+                            "runs-on-labels": platformLabels[arch] || [inputs["runs-on"]]
+                        }
+                        rockMetas.push(meta)
+                        if (isPullRequest && changesPaths.includes(`${rockPath}/rockcraft.yaml`)) {
+                            changedMetas.push(meta)
+                        } else if (!isPullRequest) {
+                            changedMetas.push(meta)
+                        }
+                    }
+                } else {
+                    const image = `${{inputs.registry}}/${{inputs.owner}}/${rockName}:${fileHash}`
+                    core.info(`generate image name: ${image}`)
+                    images.push(image)
+                    let meta = {
+                        name: rockName,
+                        version: rockVersion,
+                        path: rockPath,
+                        arch: defaultArch,
+                        image: image,
+                        "rockcraft-revision": rockcraftRevisions[defaultArch] || '',
+                        "runs-on-labels": platformLabels[defaultArch] || [inputs["runs-on"]]
+                    };
+                    rockMetas.push(meta)
+                    if (isPullRequest && changesPaths.includes(`${rockPath}/rockcraft.yaml`)) {
+                        changedMetas.push(meta)
+                    } else if (!isPullRequest) {
+                        changedMetas.push(meta)
+                    }
                 }
-              } else {
-                const image = `${{ inputs.registry }}/${{ inputs.owner }}/${rockName}:${fileHash}`
-                core.info(`generate image name: ${image}`)
-                images.push(image)
-                rockMetas.push({
-                  name: rockName,
-                  version: rockVersion,
-                  path: rockPath,
-                  arch: defaultArch,
-                  image: image,
-                  "rockcraft-revision": rockcraftRevisions[defaultArch] || '',
-                  "runs-on-labels": platformLabels[defaultArch] || [inputs["runs-on"]]
-                })
-              }
-              rockPaths.push(rockPath)
+                rockPaths.push(rockPath)
             }
             core.setOutput('rock-metas', JSON.stringify(rockMetas))
+            core.setOutput('changed-metas', JSON.stringify(changedMetas))
             core.setOutput('rock-paths', JSON.stringify(rockPaths))
             core.setOutput('images', JSON.stringify(images))
 
   build-rocks:
     name: Build rock
     needs: [get-rocks]
-    if: ${{ needs.get-rocks.outputs.rock-metas != '[]' }}
+    if: ${{ needs.get-rocks.outputs.changed-metas != '[]' }}
```

Comment:
please rename this to something like `changed-rock-metas` to indicate to which kind of metas you are referring.